aten/src/ATen/core/ivalue.h

```diff
@@ -731,6 +731,12 @@ inline bool IValue::isSameIdentity(IValue& rhs) {
   } else if (this->isBool() && rhs.isBool()) {
     // for bool type, do equality check
     return this->toBool() == rhs.toBool();
+  } else if (this->isInt() && rhs.isInt()) {
```

Comment:
The comment above clarifies that `1 is (100-99)` is not necessisarily true, and for consistency we should just return false for comparisons among integers.